pages/_app.js

```diff
@@ -1,10 +1,18 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
 
 export default function App({ Component, pageProps }) {
```

Comment:
Missing 2 further props.
```suggestion
export default function App({ Component, pageProps, setProduct, productToBuy }) {
```